functions/src/modules/TE.functions.ts

```diff
@@ -37,45 +35,48 @@ export const processAllotment = functions.https.onCall(
       .once('value');
     if (!userRef.exists()) {
       throw new functions.https.HttpsError(
-        'not-found',
-        "Assignee wasn't found!."
+        'invalid-argument',
+        "Assignee wasn't found!"
       );
     }
 
     // Update the task
-    const allTasks = [];
-    tasks.forEach(async taskId => {
-      const regex = /(\w+)-(\d+)-(\d+)/g;
-      const taskIdMatch = regex.exec(taskId);
-      const list = taskIdMatch[1];
+    const allTasks = await Promise.all(
+      tasks.map(async (taskId: string) => {
+        const list = helpers.extractListFromFilename(taskId);
 
-      await db.ref(`/edited/${list}/${taskId}/trackEditing`).update({
-        status: 'Given',
-        assignee: assignee,
-        givenTimestamp: moment().format('MM/DD/YYYY'),
-      });
+        const teDbRef = db.ref(`/edited/${list}/${taskId}/trackEditing`);
```

Comment:
I'd call it `taskRef`.
And one thought stroke me just now — it would be good to test that all the tasks are really found in the database and are spare! I'm adding this to the card description.